bot/exts/fun/movie.py

```diff
@@ -71,28 +79,34 @@ async def movies(self, ctx: Context, genre: str = "", amount: int = 5) -> None:
         # Capitalize genre for getting data from Enum, get random page, send help when genre don't exist.
         genre = genre.capitalize()
         try:
-            result = await self.get_movies_data(self.http_session, MovieGenres[genre].value, 1)
+            result, status = await self.get_movies_data(self.http_session, MovieGenres[genre].value, 1)
         except KeyError:
             await invoke_help_command(ctx)
             return
 
         # Check if "results" is in result. If not, throw error.
         if "results" not in result:
             err_msg = (
-                f"There is problem while making TMDB API request. Response Code: {result['status_code']}, "
-                f"{result['status_message']}."
+                f"There was a problem making the TMDB API request. Response Code: {status}, "
+                f"TMDB: Status Code: {result.get('status_code', None)} "
+                f"TMDB: Status Message: {result.get('status_message', None)}, "
+                f"TMDB: Errors: {result.get('errors', None)}, "
             )
             await ctx.send(err_msg)
             logger.warning(err_msg)
```

Comment:
This doesn't raise an error, so if this block was hit, code execution would continue and try to access `result` further down.

To aid with this, you could raise a `bot.utils.exceptions.APIError` with this message, and the bot's error handler will handle the sending of the error for you.

See the wikipedia cog for how this is used if needed.

Also, we should move this error handling into `get_movies_data` function, as this removes both code duplication and the need to pass around the status code.

Reply:
Alright. But I think as you suggested, I'll have to see the wikipedia cog to move forward.
(I'll try to do this in a couple of weeks as I have my exams from next week)

Reply:
No problem, real life come first as always!

Take your time and let us know if #dev-contrib if you need help!

Reply:
Hey @ChrisLovering 👋🏻. I've implemented this, please review whenever possible.